js/style/style.js

```diff
@@ -370,7 +370,10 @@ Style.prototype = util.inherit(Evented, {
         layer = this.getReferentLayer(layer);
         layer.setLayoutProperty(name, value);
         this._broadcastLayers();
-        this.sources[layer.source].reload();
+        if (layer.source) {
```

Comment:
Right, ok. Can you:

* Make the same change for `setPaintProperty`
* Add tests for `setPaintProperty` and `setLayoutProperty` on a background layer